### DISCUSSION TITLE
Batch commands

### DISCUSSION THREAD

**Discussion by Lalikinelo** [semiautomaticgit/SemiAutomaticClassificationPlugin#171] [Q&A] [answered]
Hi,

I'm a new SCP user. Thanks for this usefull plugin!

2 questions about batch commands:

1. Classification works fine but this post-processing command doesn't work:

`_classification_sieve;input_raster_path : '!working_dir!/classificationsentinel2.tif';size_threshold : 2;pixel_connection : 4;output_raster_path : '!working_dir!/classificationsentinel2clean.tif'_`

no error, nothing happend. I don't what's wrong and how to debug.

2. This command doesn't pass the check, I think that the " caracter is the problem. Is there a workaroud?

`_qgis_processing;command : 'native:fieldcalculator';parameters : 'input':'!working_dir!/classificationsentinel2final.gpkg','flield_name':'indice_refresh','field_type':'float32','field_length':'5','field_precision':'5','formula': 'CASE WHEN "DN"=1 THEN 1 WHEN "DN"=2 THEN 0.8 WHEN "DN"=3 THEN 0.7 WHEN "DN"=4 THEN 0  WHEN "DN"=5 THEN 0.3  WHEN "DN"=6 THEN 0.12 END';load_results : 1_`

Thanks for your answer!

**Comment by semiautomaticgit** [comment, marked as answer]
Hello,
about your first issue it is related to the path. It seems that input_raster_path is not correct or the input raster is not found.

About the second issue, yes it is a " issue. You should escape the " such as \\"DN\\" .
Also there is a typo, 'flield_name' should be replaced by 'field_name' in your expression.

For instance this works for me (there is an error message about GDAL that can be ignored)
`qgis_processing;command : 'native:fieldcalculator' ;parameters :'FIELD_LENGTH' : 3, 'FIELD_NAME' : 'newfield', 'FIELD_PRECISION' : 3, 'FIELD_TYPE' : 0, 'FORMULA' : 'CASE WHEN \"DN\"=1 THEN 1 WHEN \"DN\"=2 THEN 0.9 END', 'INPUT' : 'C:/Users/user/Desktop/shape.shp', 'OUTPUT' : 'C:/Users/user/Desktop/output.gpkg';load_results : 1`
